refactor: Update rest api spec

mpemy opened the pull request:
<base target="_blank">
To review use the links and instructions below and provide feedback by starting a new comment on this pull request.

- [Catalog REST API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/catalog.yaml)
- [Profile REST API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/profile.yaml)
- [System Security Plan REST API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/ssp.yaml)
- [Component REST API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/component.yaml)
- [Assessment Plan REST API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/ap.yaml)
- [Assessment Results REST API](https://editor.swagger.io//?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/ar.yaml)
- [POAM REST API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/poam.yaml)

 

Due to file size limitations with GitHub, in order to view the complete specification, please do the following:
1. Download [THIS FILE](https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/restapi.yaml) to your local machine
2. Visit [Swagger editor](https://editor-next.swagger.io) 
3. Click file, open,

**Please provide any feedback or comments by July 31.**

Commit:
First draft of the REST API, SSP routes have been added

Commit:
Adding other paths to the SSP mmodel

Commit:
Fixing few issues with the current metadata routes and add new assemblies to the system implementation

Commit:
Add Catalog Metadata routes

Commit:
Adding routes for AR POAM and AP

Commit:
Error Message objects added

Commit:
Adding new error message

Commit:
Fixing few routes in Metadata

Commit:
Adding few more assemblies

Commit:
Adding new routes to various models

Commit:
Fixing some issues with collections

Commit:
SSP System Implementation routes added

Commit:
Adding more SSP Routes

Commit:
Some more routes

Commit:
More routes

Commit:
More catalog  routes

Commit:
more catalog routes

Commit:
Adding parts

Commit:
Adding sub control routes

Commit:
Done with Catalogs

Commit:
Latest version

Commit:
Assessment Plan

Commit:
component def updated

Commit:
Components done

Commit:
updated rest API file

Commit:
updates

Commit:
Fixing commit issue

Commit:
Adding the remaining OSCAL assemblies

Commit:
Fixing linting issues

Commit:
Adding profile, catalog and ssp assemblies

Commit:
Fixing linting issues

Commit:
Adding missing assemblies for AP

Commit:
Assessment result Model completed

Commit:
All models now resolved fully.

Commit:
Add oauth labels

Commit:
few changes to openapi file

Commit:
Fixing failing build

Commit:
Adding basic and complete rest API files

Commit:
Adding attachments and snapshots

Commit:
Fix the linting issues

Commit:
Addressing the remaining issues with resolved profile

Commit:
Fixing snapshot and rsolved profiles

Commit:
All issues should be fixed

Commit:
Fixing GET model files

Commit:
Replacing file-uuid by file-id

Commit:
Addressing editing issues

Commit:
Last Update

Commit:
Final update

Commit:
Fixing snapshot

Commit:
Latest version with UUID

laurelmay commented on basicRestapi.yaml:
```diff
@@ -0,0 +1,12008 @@
+    All updates to 'root' OSCAL objects via `PATCH` requests should accept a partial payload containing
+    only the changed data and the entire updated object should be returned.
+
+    The UUID of the root object can only be changed in the server-side implementation, any attempt to change the UUID 
+    of the root object in the client-side implementation should throw an error. 
```
Let's try to be a little more specific with RFC 2119-like language here maybe?

> The document (root)-level `uuid` field and `metadata.last-modified` fields have
> specific semantics in the OSCAL specifications. Client implementations of this
> API specification must not modify the `uuid` or `metadata.last-modified` fields
> as part of a `PATCH` request. Server implementations may return an error when
> a client includes these fields in a request or they may ignore those attributes.
> As with all modifications, if the server elects to ignore the `uuid` and
> `metadata.last-modified` field in the request body, it must generate a new UUID
> and appropriately set the `last-modified` field in the document. These newly
> calculated values must be included in the response as part of the document content.

Commit:
UUID fixed

Commit:
Fix failing build

Commit:
Addressing Post Return

Commit:
Some changes

Commit:
update REST API

Commit:
Fixing Attachment_list

Commit:
Addressing the remaining issues

Commit:
Add oscal.io ref

Commit:
adding yaml support

brian-ruf-ezd committed:
changed version 0.2.0 to 0.2.1 (#75)

The Swagger capability I'm using to publish the REST API specification
requires the version number be incremented. In fairness, this is a minor
update to the 0.2.0 version we published a few month ago. Thus I am
changing the version to 0.2.1 in the `basicRestapi.yaml` file.

Commit:
addressing cleanup issues 76, 90 and 79

Commit:
removing  models API and only adding OSCALRestOpenAPI

Commit:
addressing issue 90

Commit:
fixing the linting issue

Commit:
fixing punctuation

mpemy committed:
Merge branch 'develop' into RefactorRestAPI

Commit:
fix issue 91

Commit:
addressing issue 88

Commit:
addressing issue 81

Commit:
addressing issue 87

Commit:
issue 87

Commit:
completing issue 91

Commit:
completing issues 91 and 76

brian-ruf-ezd committed:
reduction script in Python (#92)

Python code to remove the unused schema objects.

Commit:
issue 89

Commit:
fixing linting issues

Commit:
issue 84

Commit:
issue 84 again

Commit:
fix issues 84 and 85.

Commit:
fix issues 84 and 85 and updated version.

Commit:
issue 85.

Commit:
issue 83

Commit:
issue 85 fixed

Commit:
issue 77

Commit:
fixing issue 77

Commit:
handling attachment

Commit:
handling attachment fix

Commit:
attachment latest fix

Commit:
xml wrapper fix

Commit:
xml wrapper

brian-ruf-ezd committed:
Merge branch 'develop' into RefactorRestAPI

brian-ruf-ezd reviewed (approved):
I've reviewed all of this content using an [OpenAPI viewer](https://raw.githack.com/EasyDynamics/oscal-rest/develop/viewer/index.html?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/RefactorRestAPI/OSCALRestOpenAPI.yaml).
